examples/fabric-admin/device_manager/DeviceSynchronization.cpp

```diff
@@ -72,6 +72,13 @@ void DeviceSynchronizer::OnAttributeData(const ConcreteDataAttributePath & path,
     VerifyOrDie(path.mEndpointId == kRootEndpointId);
     VerifyOrDie(path.mClusterId == Clusters::BasicInformation::Id);
 
+    CHIP_ERROR error = status.ToChipError();
```

Comment:
Can you help me understand something: Is this an issue for wildcard reads of the BINFO cluster? If so I would like to understand what changed, because this was working without issues a couple weeks ago even for commissioning a device that didn't have a UID.

Is this status set to an error only for the single attribute of the wild card read, or for all of the attributes

Reply:
This is a regression after we refactor to DeviceSynchronization. 

See we have this check in:
https://github.com/project-chip/connectedhomeip/blob/master/examples/fabric-admin/commands/clusters/ReportCommand.cpp#L30

Reply:
Which refactor of `DeviceSynchronization` are you referring to? We had `DeviceSynchronization` in TE2 where we passed MCORE_1_3 (at the time of TE2 I am pretty sure Andrei commissioned TH_SERVER with no UID and was able to validate that DUT created a UID on TH_SERVERs behalf

Reply:
I am taking about OnAttributeData in DeviceSynchronization, do we read UID from the TH_SERVER without UID in TE2?

From the current implementation, it should 100% crash if we try to decode TLV data if status is not success.

```
void DeviceSynchronizer::OnAttributeData(const ConcreteDataAttributePath & path, TLV::TLVReader * data, const StatusIB & status)
{
    VerifyOrDie(path.mEndpointId == kRootEndpointId);
    VerifyOrDie(path.mClusterId == Clusters::BasicInformation::Id);

    switch (path.mAttributeId)
    {
    case Clusters::BasicInformation::Attributes::UniqueID::Id:
        mCurrentDeviceData.has_unique_id =
            SuccessOrLog(data->GetString(mCurrentDeviceData.unique_id, sizeof(mCurrentDeviceData.unique_id)), "UniqueId");
        break;
```